src/java.base/share/classes/jdk/internal/access/SharedSecrets.java

```diff
@@ -60,6 +61,9 @@ interface and provides the ability to call package-private methods
  * </strong>
  */
 
+// Static fields in this class are stateless, so the values initialized in the
+// AOT assembly phase can be safely cached.
+@AOTSafeClassInitializer
 public class SharedSecrets {
     // This field is not necessarily stable
     private static JavaAWTFontAccess javaAWTFontAccess;
```

Comment:
Does aot initialization work with this field?

Reply:
I think this field is safe. There are two places that could set it, but they will always set it to an instance of `JavaAWTFontAccessImpl`, which is stateless.

https://github.com/openjdk/jdk/blob/eff6439e75d79c67370e79638024296e01101b48/src/java.desktop/share/classes/java/awt/font/NumericShaper.java#L148-L149

https://github.com/openjdk/jdk/blob/eff6439e75d79c67370e79638024296e01101b48/src/java.desktop/share/classes/java/awt/font/TextAttribute.java#L251-L252